Sources/Web3Modal/Web3ModalClient.swift

```diff
@@ -0,0 +1,160 @@
+    ///
+    /// Event will be emited on controller and non-controller clients.
+    public var sessionsPublisher: AnyPublisher<[Session], Never> {
+        signClient.sessionsPublisher.eraseToAnyPublisher()
+    }
+    
+    /// Publisher that sends web socket connection status
+    public var socketConnectionStatusPublisher: AnyPublisher<SocketConnectionStatus, Never> {
+        signClient.socketConnectionStatusPublisher.eraseToAnyPublisher()
+    }
+    
+    /// Publisher that sends session when one is settled
+    ///
+    /// Event is emited on proposer and responder client when both communicating peers have successfully established a session.
+    public var sessionSettlePublisher: AnyPublisher<Session, Never> {
+        signClient.sessionSettlePublisher.eraseToAnyPublisher()
+    }
+    
+    /// Publisher that sends session proposal that has been rejected
+    ///
+    /// Event will be emited on dApp client only.
+    public var sessionRejectionPublisher: AnyPublisher<(Session.Proposal, Reason), Never> {
+        signClient.sessionRejectionPublisher.eraseToAnyPublisher()
+    }
+    
+    /// Publisher that sends deleted session topic
+    ///
+    /// Event can be emited on any type of the client.
+    public var sessionDeletePublisher: AnyPublisher<(String, Reason), Never> {
+        signClient.sessionDeletePublisher.eraseToAnyPublisher()
+    }
+    
+    /// Publisher that sends response for session request
+    ///
+    /// In most cases that event will be emited on dApp client.
+    public var sessionResponsePublisher: AnyPublisher<Response, Never> {
+        signClient.sessionResponsePublisher.eraseToAnyPublisher()
+    }
+    
+    // MARK: - Private Properties
+    private let signClient: SignClientProtocol
+    private let pairingClient: (PairingClientProtocol & PairingInteracting)
+    
+    init(
+        signClient: SignClientProtocol,
+        pairingClient: (PairingClientProtocol & PairingInteracting)
+    ) {
+        self.signClient = signClient
+        self.pairingClient = pairingClient
+    }
+    
+    /// For creating new pairing URI
+    public func createPairing() async throws -> WalletConnectURI {
+        try await pairingClient.create()
+    }
+    
+    /// For proposing a session to a wallet.
+    /// Function will propose a session on existing pairing.
+    /// - Parameter topic: topic from existing pairing.
+    public func connect(on topic: String) async throws {
+        try await signClient.connect(
+            requiredNamespaces: Web3Modal.config.sessionParams.requiredNamespaces,
+            optionalNamespaces: Web3Modal.config.sessionParams.optionalNamespaces,
+            sessionProperties: Web3Modal.config.sessionParams.sessionProperties,
+            topic: topic
+        )
+    }
+    
+    /// For proposing a session to a wallet.
+    /// Function will propose a session on newly created pairing.
+    public func createPairingAndConnect() async throws -> WalletConnectURI {
```

Comment:
would make sense to call the method just connect() with an optional topic?
same as deprecated method of sign client

```
    public func connect(
        requiredNamespaces: [String: ProposalNamespace],
        optionalNamespaces: [String: ProposalNamespace]? = nil,
        sessionProperties: [String: String]? = nil,
        topic: String? = nil
```

Reply:
actually would match kotlin's interface too